packages/e2e-cypress/src/helpers/cct-dev-server/index.ts

```diff
@@ -42,54 +34,80 @@ async function exportQuasarConfig(bundler: AvailableBundlers): Promise<any> {
   // register app extensions
   await extensionRunner.registerExtensions(ctx);
 
-  if (bundler === 'vite') {
-    const quasarConfFile = new QuasarConfFile({ ctx });
+  return {
+    quasarAppPackage,
+    QuasarConfFile,
+    ctx,
+  };
+}
 
-    const quasarConf = await quasarConfFile.read();
-    if (quasarConf.error !== void 0) {
-      console.log(quasarConf.error);
-    }
+async function quasarWebpackConfig(quasarAppPackage: string) {
+  const { QuasarConfFile, ctx } = await quasarSharedConfig(
+    quasarAppPackage,
+    'webpack',
+  );
+
+  const {
+    default: { splitWebpackConfig },
+  } = await import(`${quasarAppPackage}/lib/webpack/symbols`);
 
-    const { default: generateConfig } = await import(
-      `${quasarAppPackage}/lib/modes/spa/spa-config`
-    );
+  const quasarConfFile = new QuasarConfFile(ctx);
 
-    return generateConfig['vite'](quasarConf);
-  } else {
-    const {
-      default: { splitWebpackConfig },
-    } = await import(`${quasarAppPackage}/lib/webpack/symbols`);
+  try {
+    await quasarConfFile.prepare();
+  } catch (e) {
+    console.log(e);
+    return;
+  }
+  await quasarConfFile.compile();
 
-    const quasarConfFile = new QuasarConfFile(ctx);
+  const configEntries = splitWebpackConfig(quasarConfFile.webpackConf, 'spa');
 
-    try {
-      await quasarConfFile.prepare();
-    } catch (e) {
-      console.log(e);
-      return;
-    }
-    await quasarConfFile.compile();
+  return configEntries[0].webpack;
+}
 
-    const configEntries = splitWebpackConfig(quasarConfFile.webpackConf, 'spa');
+async function quasarViteConfig(quasarAppPackage: string) {
+  const { QuasarConfFile, ctx } = await quasarSharedConfig(
+    quasarAppPackage,
+    'vite',
+  );
 
-    return configEntries[0].webpack;
+  const quasarConfFile = new QuasarConfFile({ ctx });
+
+  const quasarConf = await quasarConfFile.read();
+  if (quasarConf.error !== void 0) {
+    console.log(quasarConf.error);
   }
+
+  const { default: generateConfig } = await import(
+    `${quasarAppPackage}/lib/modes/spa/spa-config`
+  );
+
+  return generateConfig['vite'](quasarConf);
```

Comment:
```suggestion
  // [1] -> https://github.com/cypress-io/cypress/issues/22505#issuecomment-1277855100
  // [1] Make sure to use the root for predictability
  quasarConf.publicPath = '/';

  const result = await generateConfig['vite'](quasarConf);

  // [1] Delete base so it can correctly be set by Cypress
  delete result.base;

  return result;
```